### PR TITLE
Gzip and deflate encoding support for http client curl

### PR DESCRIPTION
**Pull request by milchakov**

### CODE REVIEW

**Comment by milchakov** [comment]
DUDE JTTP

File: platform/http_client_curl.cpp [hunk 1]
```diff
@@ -144,6 +148,18 @@ bool WriteToFile(string const & fileName, string const & data)
   ofs << data;
   return true;
 }
+
+std::string Decompress(std::string const & compressed, std::string const & encoding)
+{
+  std::string decompressed;
+
+  if (it.second == "gzip" || it.second == "deflate")
```
Comment (ygorshenin) [inline platform/http_client_curl.cpp]: Could you please explain where it is declared?

Comment (ygorshenin) [inline platform/http_client_curl.cpp]: Please, compile the pull request first.

**Comment by milchakov** [comment]
DUDE JTTP

File: platform/http_client_curl.cpp (hunk 1 above)
Comment (milchakov) [inline platform/http_client_curl.cpp]: не могу

**Comment by milchakov** [comment]
DUDE JTTP

**Comment by therearesomewhocallmetim** [comment]
DUDE JTTP

**Comment by milchakov** [comment]
DUDE JTTP

**Comment by therearesomewhocallmetim** [comment]
LGTM

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**